chromadb/server/fastapi/__init__.py

```diff
@@ -290,6 +296,24 @@ def heartbeat(self) -> Dict[str, int]:
     def version(self) -> str:
         return self._api.get_version()
 
+    @trace_method("FastAPI.env", OpenTelemetryGranularity.OPERATION)
+    @authz_context(
+        action=AuthzResourceActions.ENV,
+        resource=DynamicAuthzResource(
+            type=AuthzResourceTypes.API,
+        ),
+    )
+    def env(self) -> SystemInfo:
+        _local_env = self._api.env()
+        if not _local_env or "client" not in _local_env or not _local_env["client"]:
+            raise HTTPException(
+                status_code=500,
+                detail="Unable to get system info.",
+            )
+        # this can be done better for distributed mode
+        _local_env["client"]["mode"] = OperatingMode.SERVER_SINGLE_NODE
```

Comment:
Can we find a way to do this? It feels hacky to me. I would rather we find a setting to read to determine our operating mode when we populate it.

Reply:
We can try to resolve this by detecting whether the FastAPI server is loaded, but that might be equally not more hacky. @HammadB, wdyt?

Reply:
Added a backend settings flag that FastAPI or distributed segment server sets and we can use it to detect the operating mode of Chroma in utils module